velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -157,7 +162,7 @@ FOLLY_ALWAYS_INLINE uint64_t updateTail(uint64_t hash, uint64_t value) {
   return XXH_rotl64(temp, 27) * XXH_PRIME64_1 + XXH_PRIME64_4;
 }
 
-FOLLY_ALWAYS_INLINE uint64_t hashLongDecimalPart(const uint64_t value) {
+FOLLY_ALWAYS_INLINE int64_t hashLongDecimalPart(const uint64_t value) {
```

Comment:
Why change return type ? Will it affect previously computed hashes ?

Reply:
oh oops, I commented on this in phab, but I guess it doesn't sync here. `XXH64` returns int64_t.

This was actually my exact question for folks with more context, would doing the narrow conversion here be okay?

Reply:
I am not sure and I probably think not but I would keep it as is for now.

Reply:
Yep, I am keeping it as is for now.

What I mean is `XXH64` returns uint64_t, why we sometimes down cast it to int64_t?

Reply:
Yes exactly @bikramSingh91 . This is my exact question, apologies to Krishna for being unclear.

In java, the hashcode is a long (which means it can be negative, see https://github.com/prestodb/presto/blob/04138bc26a4fc801630fe936463d89d071d00c0a/presto-main/src/main/java/com/facebook/presto/type/IpAddressOperators.java#L111-L114)


Let me do more testing before landing this